### ISSUE TITLE
[bug] Provider affects the DOM,

### ISSUE THREAD

**Issue by LufyCZ** [rainbow-me/rainbowkit#2164]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

2.1.5

### wagmi Version

2.17.0

### Current Behavior

The `RainbowKitProvider` component injects a `<div>` which affects styling.

![image](https://github.com/user-attachments/assets/d9facc79-dbde-4bc4-8643-2a44f85cc0b6)

The above should result in a fully red screen, however, the "injected" div doesn't have a `h-full`, and thus the div below won't stretch to the height of the screen.

![image](https://github.com/user-attachments/assets/8c964603-4f05-4938-be90-c6d202cdce17)

### Expected Behavior

The provider should not affect the DOM in a way that potentially breaks styling.

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://github.com/LufyCZ/rainbowkit-bug

**Comment by Adebesin-Cell** [comment]
You can try. 

```tsx
function MyApp() {
  return (
    <WagmiProvider config={config}>
      <QueryClientProvider client={client}>
        <RainbowKitProvider>
          <div style={{ height: "100vh" }}>
            <div style={{ height: "100%", background: "red" }}>
              <p>lorem ipsum</p>
            </div>
          </div>
        </RainbowKitProvider>
      </QueryClientProvider>
    </WagmiProvider>
  );
}

```

**Comment by LufyCZ** [comment]
> You can try.
> 
> ```tsx
> function MyApp() {
>   return (
>     <WagmiProvider config={config}>
>       <QueryClientProvider client={client}>
>         <RainbowKitProvider>
>           <div style={{ height: "100vh" }}>
>             <div style={{ height: "100%", background: "red" }}>
>               <p>lorem ipsum</p>
>             </div>
>           </div>
>         </RainbowKitProvider>
>       </QueryClientProvider>
>     </WagmiProvider>
>   );
> }
> ```

I solved it with `[&>]:h-full` on the div right above the `RainbowKitProvider`.

However, what you are proposing is still not a "correct" solution. The provider could be used somewhere further down the tree, and, for example, if there was a Header above it, the page would then be `px(Header) + 100vh` high (which is more than 100vh).

**Comment by Adebesin-Cell** [comment]
> > You can try.
> > ```tsx
> > function MyApp() {
> >   return (
> >     <WagmiProvider config={config}>
> >       <QueryClientProvider client={client}>
> >         <RainbowKitProvider>
> >           <div style={{ height: "100vh" }}>
> >             <div style={{ height: "100%", background: "red" }}>
> >               <p>lorem ipsum</p>
> >             </div>
> >           </div>
> >         </RainbowKitProvider>
> >       </QueryClientProvider>
> >     </WagmiProvider>
> >   );
> > }
> > ```
> 
> I solved it with `[&>*]:h-full` on the div right above the `RainbowKitProvider`.
> 
> However, what you are proposing is still not a "correct" solution. The provider could be used somewhere further down the tree, and, for example, if there was a Header above it, the page would then be `px(Header) + 100vh` high (which is more than 100vh).

Yeah, good point! Though I did add a fix to clone the child element and add the props to it, got closed haha.

**Comment by minato32** [comment]
I'd like to work on this. Will open a PR.

**Comment by SnowingFox** [comment]
Opened a PR to address this: https://github.com/rainbow-me/rainbowkit/pull/2693
